### PR TITLE
HIVE-25321: advance write id during AlterTableDropPartition

### PR DESCRIPTION
**Pull request by kishendas**
### Why are the changes needed?
Advance Write ID during AlterTableDropPartition DDL

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Unit tests

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java [hunk 1]
```diff
@@ -105,11 +106,18 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
 
     AlterTableDropPartitionDesc desc =
         new AlterTableDropPartitionDesc(tableName, partitionSpecs, mustPurge, replicationSpec);
+
+    Task<DDLWork> ddlTask = TaskFactory.get(new DDLWork(getInputs(), getOutputs(), desc));
```
Comment (hsnusonic) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java:110]: Is `ddlTask` used?

**Comment by sourabh912** [comment]
Looks good to me.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java (hunk 1 above)
Comment (kishendas) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java]: Not used. Fixed it.

**Comment by kishendas** [comment]
It's failing during "Archiving" stage with below error, which seems to be intermittent. Hopefully I will get a green run in the next one. 

[2021-07-15T06:00:07.621Z] java.net.ProtocolException: Expected HTTP 101 response but was '500 Internal Server Error'
[2021-07-15T06:00:07.621Z] 	at okhttp3.internal.ws.RealWebSocket.checkResponse(RealWebSocket.java:229)
[2021-07-15T06:00:07.621Z] 	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:196)
[2021-07-15T06:00:07.621Z] 	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
[2021-07-15T06:00:07.621Z] 	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
[2021-07-15T06:00:07.621Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2021-07-15T06:00:07.621Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2021-07-15T06:00:07.621Z] 	at java.lang.Thread.run(Thread.java:748)
io.fabric8.kubernetes.client.KubernetesClientException: error dialing backend: EOF

**Review by hsnusonic — APPROVED** [review]

**Review by sourabh912 — APPROVED** [review]
The change looks good to me.

**Comment by kishendas** [comment]
@nrg4878 One unrelated test is failing now http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2465/12/tests . Please commit this change, whenever you get time.

**Comment by kishendas** [comment]
@nrg4878 Got a green build -> http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-2465/13/pipeline/ .

**Comment by nrg4878** [comment]
fix has been committed to master. Please close the PR. Thanks

**Pull request closed without merging**